### PR TITLE
Restructured servicemonitors for easy onboard

### PR DESCRIPTION
**Pull request by bamachrn**
This PR is for making onboarding to monitoring easy for the teams.
* restructuring servicemonitors files in the directory to make sure nothing else is changed while adding new services
* adding metrics-reader rolebinding along with the servicemonitors to make sure the prometheus has access to the metrics
* updated Readme to match the restructure 
Signed-off-by: Bama Charan Kundu <bamachrn@gmail.com>

### CODE REVIEW

File: components/monitoring/README.md [hunk 1]
```diff
@@ -70,36 +70,38 @@ spec:
       
 
   - If the servicemonitor is for getting other components added to prometheus monitoring
-      
+      - create a new file under [components/monitoring/prometheus/base/servicemonitors](https://github.com/redhat-appstudio/infra-deployments/blob/main/components/monitoring/prometheus/base/servicemonitors)
       - Add a sevicemonitor decalration for scraping the intended service
```
Comment (xcoulon) [inline components/monitoring/README.md:74]:
```suggestion
      - Add a ServiceMonitor along with a ClusterRoleBinding to scrape the intended service
```

**Review by xcoulon — APPROVED** [review]
looks good, just left a suggestion to improve the docs

File: components/monitoring/README.md (hunk 1 above)
Comment (bamachrn) [reply components/monitoring/README.md]: done

File: components/monitoring/prometheus/base/kustomization.yaml [hunk 2]
```diff
@@ -4,9 +4,9 @@ resources:
 - install-prometheus.yaml
 - prometheus-view.yaml
 - configure-prometheus.yaml
-- prometheus-servicemonitors.yaml
+- servicemonitors/
```
Comment (Michkov) [inline components/monitoring/prometheus/base/kustomization.yaml:7]: you need to include each file or have servicemonitors/kustomization.yaml (where still all files must be listed). Now kustomization is failing:
```
$ kustomize build components/monitoring/prometheus/base/
Error: accumulating resources: accumulation err='accumulating resources from 'servicemonitors/': '/home/mkovarik/git/appstudio/infra-deployments/components/monitoring/prometheus/base/servicemonitors' must resolve to a file': couldn't make target for path '/home/mkovarik/git/appstudio/infra-deployments/components/monitoring/prometheus/base/servicemonitors': unable to find one of 'kustomization.yaml', 'kustomization.yml' or 'Kustomization' in directory '/home/mkovarik/git/appstudio/infra-deployments/components/monitoring/prometheus/base/servicemonitors'
```

File: components/monitoring/prometheus/base/servicemonitors/dora-service.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      app: exporter
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: prometheus-dora-metrics-reader
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: dora-metrics-reader
```
Comment (Michkov) [inline components/monitoring/prometheus/base/servicemonitors/dora-service.yaml:32]: where is this `dora-metrics-reader` clusterRole defined?

Comment (amfred) [reply components/monitoring/prometheus/base/servicemonitors/dora-service.yaml:32]: @bamachrn I was waiting on the same answer.

Comment (bamachrn) [reply components/monitoring/prometheus/base/servicemonitors/dora-service.yaml:32]: This is from the respective repo. I looked in the dora-metrics repo 
https://github.com/redhat-appstudio/dora-metrics/blob/main/deploy/base/clusterrole.yaml
https://github.com/redhat-appstudio/dora-metrics/blob/main/deploy/base/clusterrolebinding.yaml

Comment (bamachrn) [reply components/monitoring/prometheus/base/servicemonitors/dora-service.yaml:32]: @amfred @Michkov ^

**Review by Michkov — APPROVED** [review]

File: components/monitoring/prometheus/base/servicemonitors/dora-service.yaml (hunk 3 above)
Comment (Michkov) [reply components/monitoring/prometheus/base/servicemonitors/dora-service.yaml:32]: Right, thanks

**Review by amfred — APPROVED** [review]
/lgtm

**Pull request merged**